packages/dataviews/src/components/dataform-context/index.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { FormField, NormalizedField } from '../../types';
+
+type DataFormContextType< Item > = {
+	getFieldDefinition: (
+		field: FormField
+	) => NormalizedField< Item > | undefined;
+};
+
+const DataFormContext = createContext< DataFormContextType< any > >( {
+	getFieldDefinition: () => undefined,
+} );
+
+export function DataFormProvider< Item >( {
+	fields,
+	children,
+}: React.PropsWithChildren< { fields: NormalizedField< Item >[] } > ) {
+	const getFieldDefinition = useCallback(
+		( field: FormField ) => {
+			const fieldId = typeof field === 'string' ? field : field.id;
+
+			const definition = fields.find(
+				( fieldDefinition ) => fieldDefinition.id === fieldId
+			);
+			if ( definition ) {
+				return {
+					...definition,
+					label:
+						typeof field !== 'string' && field.label
+							? field.label
+							: definition.label,
+				};
```

Comment:
I added the `label` option to the `FormField` definition, which overwrites the field definition label if provided.
I did this for two reasons:
- This way the individual layouts won't need to worry about this
- The regular layout doesn't handle the label rendering and renders the Edit component directly, which would require us to overwrite the label there anyway.